src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -196,6 +196,9 @@ export class EvaluationPlugin extends UIPlugin {
       case "EVALUATE_CELLS":
         this.evaluator.evaluateAllCells();
         break;
+      case "REMOVE_MERGE":
```

Comment:
I think the "right" approach would be to add "REMOVE_MERGE" to. the set `invalidateEvaluationCommands` which already contains ADD_MERGE, it follows the same invalidation logic IMHO

About this @anhe-odoo or @laa-odoo is there a reason why  force `shouldRebuildDependenciesGraph` to true when handling `CREATE_SHEET` even though it's already a part of `invalidateEvaluationCommands`? see 6 lines above this

Reply:
I agree with the idea that ADD_MERGE should be in the `invalidateEvaluationCommands` set. About CREATE_SHEET, it clearly make a "double emploi" here. But is there any reason for the `CREATE_SHEET` command to be in the `invalidateEvaluationCommands` set ? I tried to remove it, and the only test that failed was for the composer, so I'm not sure

Reply:
I think the mistake was to put the check on 'invalidateEvaluationCommands'  in allowDispatch, which is bypassed in some cases (namely, I think it's not considered on subcommands and on remote commands from the server). Moving the check to dispatch instead should do the trick?

Reply:
My mistake, `invalidateEvaluationCommands` are handled in beforeHandle, not allowDispatch so it's ok.

Considering the tests that only fail in the composer, it then could simply be a lack of test coverage, which we could take the opportunity to add with this pull request.

Reply:
Why did the composer-related tests pass for me? 🤔